rust/lance-index/src/scalar/btree.rs

```diff
@@ -1534,6 +1534,7 @@ impl ScalarIndexPlugin for BTreeIndexPlugin {
         data: SendableRecordBatchStream,
         index_store: &dyn IndexStore,
         request: Box<dyn TrainingRequest>,
+        _fragment_ids: Option<Vec<u32>>,
```

Comment:
I think for the other indexes, instead of just ignoring, we should fail with not supported if this is set